### ISSUE TITLE
autocompleteAddParentheses causes front(|) to occur due to const/non-const overloads

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#7540]
Thanks for the fix. In using the feature, I found another two bugs:
```
vector<int> v;
v.empty()|; // Autocomplete 'empty' to "empty()", and the cursor is after ')', this is expected.
v.front(|); // Autocomplete 'front' to "front()", but the cursor is between '(' and ')', this is unexpected, because "front" has no argument.

set<int> s;
s.insert // Expect auto adding '()' for 'insert' call.
```
Please take a look, thanks a lot for your work.

_Originally posted by @xiray in https://github.com/microsoft/vscode-cpptools/issues/7209#issuecomment-841673072_

**Comment by xiray** [comment]
Thanks for the quick fix.
I just verified the fix in v1.4.0-insiders3 release, the 1st bug has been fixed, but the 2nd bug still exists:
```
set<int> s;
s.insert // Expect auto adding '()' for 'insert' call.
```
Please take a look?

**Comment by sean-mcmanus** [comment]
Okay, I repro the issue. Investigating...

**Comment by sean-mcmanus** [comment]
Okay, those 2 cases of set::insert/vector::emplace should be fixed with 1.4.0 (when it's released).

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.4.0
